comp/core/status/statusimpl/status.go

```diff
@@ -0,0 +1,315 @@
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+)
+
+//go:embed templates
+var templatesFS embed.FS
+
+type dependencies struct {
+	fx.In
+	Config config.Component
+	Params status.Params
+
+	Providers       []status.Provider       `group:"status"`
+	HeaderProviders []status.HeaderProvider `group:"header_status"`
+}
+
+type statusImplementation struct {
+	sortedHeaderSection []status.HeaderProvider
+	sortedSections      []string
+	sortedProviders     map[string][]status.Provider
+	providers           []status.Provider
+	headerProvider      []status.HeaderProvider
+}
+
+// Module defines the fx options for this component.
+func Module() fxutil.Module {
+	return fxutil.Component(
+		fx.Provide(newStatus),
+	)
+}
+
+func sortByName(providers []status.Provider) []status.Provider {
+	sort.SliceStable(providers, func(i, j int) bool {
+		return providers[i].Name() < providers[j].Name()
+	})
+
+	return providers
+}
+
+func newStatus(deps dependencies) (status.Component, error) {
+	sections := []string{}
+
+	for _, provider := range deps.Providers {
+		if !present(provider.Section(), sections) {
+			sections = append(sections, provider.Section())
+		}
+	}
+
+	sortedProviders := map[string][]status.Provider{}
+
+	for _, provider := range deps.Providers {
+		providers := sortedProviders[provider.Section()]
+		sortedProviders[provider.Section()] = append(providers, provider)
+	}
+
+	for section, providers := range sortedProviders {
+		sortedProviders[section] = sortByName(providers)
+	}
+
+	sortedHeaderSection := deps.HeaderProviders
+	sort.SliceStable(sortedHeaderSection, func(i, j int) bool {
+		return sortedHeaderSection[i].Index() < sortedHeaderSection[j].Index()
+	})
+
+	sortedHeaderSection = append([]status.HeaderProvider{newCommonHeaderProvider(deps.Params.Name, deps.Config)}, sortedHeaderSection...)
+
+	sort.Strings(sections)
+
+	return &statusImplementation{
+		sortedSections:      sections,
+		sortedProviders:     sortedProviders,
+		sortedHeaderSection: sortedHeaderSection,
+		providers:           deps.Providers,
+		headerProvider:      deps.HeaderProviders,
+	}, nil
+}
+
+func (s *statusImplementation) GetStatus(format string, _ bool) ([]byte, error) {
+	switch format {
+	case "json":
+		stats := make(map[string]interface{})
+		for _, sc := range s.sortedHeaderSection {
+			sc.JSON(stats)
+		}
+
+		for _, providers := range s.sortedProviders {
+			for _, provider := range providers {
+				provider.JSON(stats)
+			}
+		}
+		return json.Marshal(stats)
+	case "text":
+		var b = new(bytes.Buffer)
+		var errors []error
+
+		for _, sc := range s.sortedHeaderSection {
+			if err := sc.Text(b); err != nil {
+				errors = append(errors, err)
+			}
+		}
+
+		newLine(b)
+
+		for i, sc := range s.sortedProviders[status.CollectorSection] {
+			if i == 0 {
+				printHeader(b, status.CollectorSection)
+				newLine(b)
+			}
+
+			if err := sc.Text(b); err != nil {
+				errors = append(errors, err)
+			}
+		}
```

Comment:
We could check for the header only once:

```suggestion
		if len(s.sortedProviders[status.CollectorSection]) != 0 {
			printHeader(b, status.CollectorSection)
			newLine(b)

			for _, sc := range s.sortedProviders[status.CollectorSection] {
				if err := sc.Text(b); err != nil {
					errors = append(errors, err)
				}
			}
		}
```
WDYT ?